jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/PropertyRule.java

```diff
@@ -130,22 +122,25 @@ private boolean isRequired(String nodeName, JsonNode node, Schema schema) {
         return false;
     }
 
-    private boolean useOptional(String nodeName, JsonNode node, Schema schema) {
-        if (node.has("javaOptional")) {
-            final JsonNode requiredNode = node.get("javaOptional");
+    private boolean hasFlag(JsonNode node, String fieldName) {
+        if (node.has(fieldName)) {
+            final JsonNode requiredNode = node.get(fieldName);
             return requiredNode.asBoolean();
         }
 
-        JsonNode javaOptionalArray = schema.getContent().get("javaOptional");
+        return false;
+    }
+
+    private boolean isDeclaredAs(String type, String nodeName, JsonNode node, Schema schema) {
+        return hasEnumerated(schema, type, nodeName) || hasFlag(node, type);
```

Comment:
You can change the checking order here if you feel the increasing performance here.
We can do that because `hasFlag` is returning boolean independently and if it returns `false` then we check the top `required` array. In the previous implementation, we did not check the top `required` array if the test of flag returned `false.